### DISCUSSION TITLE
authentication to cosmosdb using AAD / Service Principal instead only master.key

### DISCUSSION THREAD

**Discussion by elayaraja-kv** [microsoft/kafka-connect-cosmosdb#571] [Q&A]
Hi Team,

Noticed that from the config, as of now kafka connect to cosmosdb authentication is enabled only using "**connect.cosmos.master.key**". Is this true? If not true, please let me know the correct configuration for using Azure Service Principal.

If its true, any plan to add this feature?

**Comment by elayaraja-kv** [comment]
Looks like there is existing issue: https://github.com/microsoft/kafka-connect-cosmosdb/issues/570

**Comment by kushagraThapar** [comment]
@esquare666 - this feature has been developed however, on the v2 version of this kafka connector, and we plan to continue this development on the v2 version only. You can learn more about the new version here - https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/cosmos/azure-cosmos-kafka-connect - this new version has all the features listed here in the changelog- https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos-kafka-connect/CHANGELOG.md

**Comment by kushagraThapar** [comment]
You can read more about it here - https://devblogs.microsoft.com/cosmosdb/enhancements-in-the-kafka-connector-for-azure-cosmos-db-a-new-era-of-scalability-and-flexibility/
